integrations/google_vertex/src/haystack_integrations/components/generators/google_vertex/chat/gemini.py

```diff
@@ -355,6 +355,68 @@ def run(
 
         return {"replies": replies}
 
+    @component.output_types(replies=List[ChatMessage])
+    async def run_async(
+        self,
+        messages: List[ChatMessage],
+        streaming_callback: Optional[StreamingCallbackT] = None,
+        *,
+        tools: Optional[List[Tool]] = None,
+    ):
+        """
+        Async version of the run method. Generates text based on the provided messages.
+        :param messages:
+            A list of `ChatMessage` instances, representing the input messages.
+        :param streaming_callback:
+            A callback function that is called when a new token is received from the stream.
+        :param tools:
+            A list of tools for which the model can prepare calls. If set, it will override the `tools` parameter set
+            during component initialization.
+        :returns:
+            A dictionary containing the following key:
+            - `replies`:  A list containing the generated responses as `ChatMessage` instances.
+        """
+        streaming_callback = select_streaming_callback(
+            self._streaming_callback, streaming_callback, requires_async=True
+        )
+
+        tools = tools or self._tools
+        _check_duplicate_tool_names(tools)
+        google_tools = self._convert_to_vertex_tools(tools) if tools else None
+
+        if messages[0].is_from(ChatRole.SYSTEM):
+            self._model._system_instruction = Part.from_text(messages[0].text)
+            messages = messages[1:]
+
+        google_messages = [_convert_chatmessage_to_google_content(m) for m in messages]
+
+        session = self._model.start_chat(history=google_messages[:-1])
+
+        candidate_count = 1
+        if self._generation_config:
+            config_dict = self._generation_config_to_dict(self._generation_config)
+            candidate_count = config_dict.get("candidate_count", 1)
+
+        if streaming_callback and candidate_count > 1:
+            msg = "Streaming is not supported with multiple candidates. Set candidate_count to 1."
+            raise ValueError(msg)
+
+        res = await session.send_message_async(
+            content=google_messages[-1],
+            generation_config=self._generation_config,
+            safety_settings=self._safety_settings,
+            stream=streaming_callback is not None,
+            tools=google_tools,
```

Comment:
We should add `tool_config=self._tool_config,` similar to the implementation in the GoogleAIGeminiChatGenerator
```suggestion
            tools=google_tools,
            tool_config=self._tool_config,
```
https://github.com/deepset-ai/haystack-core-integrations/pull/1543/files#diff-6ded5a141e23817de550ebfaf77e82665b8642c3fe53091dde28c61793177d69R345